lib/cloud/imds/gcp/imds.go

```diff
@@ -76,8 +76,20 @@ func NewInstanceMetadataClient(ctx context.Context) (*InstanceMetadataClient, er
 
 // IsAvailable checks if instance metadata is available.
 func (client *InstanceMetadataClient) IsAvailable(ctx context.Context) bool {
-	instanceData, err := client.getMetadata(ctx, "instance")
-	return err == nil && instanceData != ""
+	id, err := client.getNumericID(ctx)
+	return err == nil && id != 0
+}
+
+func (client *InstanceMetadataClient) getNumericID(ctx context.Context) (uint64, error) {
+	idStr, err := client.GetID(ctx)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+	id, err := strconv.ParseUint(idStr, 10, 64)
+	if err != nil {
+		return 0, trace.Wrap(err)
```

Comment:
Let's return a better error here. Maybe something like this?

```suggestion
		return 0, trace.BadParameter("Invalid instance ID %q", idStr)
```

Otherwise the logs will say something like: "Error looking for cloud instance metadata: parsing "asdf": invalid syntax"